### PR TITLE
Update check-config.sh, netlink and libnetwork vendoring

### LINKED ISSUE #22185: DNS queries sometimes get stuck since upgrading to 1.11.0
**Output of `docker version`:**

```
Client:
 Version:      1.11.0
 API version:  1.23
 Go version:   go1.6.1
 Git commit:   9e83765
 Built:
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.0
 API version:  1.23
 Go version:   go1.6.1
 Git commit:   9e83765
 Built:
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 27
 Running: 26
 Paused: 0
 Stopped: 1
Images: 8
Server Version: 1.11.0
Storage Driver: devicemapper
 Pool Name: docker-254:4-537178368-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/data/dockerdata
 Metadata file: /dev/data/dockermetadata
 Data Space Used: 17.63 GB
 Data Space Total: 107.4 GB
 Data Space Available: 89.74 GB
 Metadata Space Used: 33.01 MB
 Metadata Space Total: 5.369 GB
 Metadata Space Available: 5.336 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Library Version: 1.03.01 (2011-10-15)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: null host bridge
Kernel Version: 3.16.7-35-default
Operating System: openSUSE 13.2 (Harlequin) (x86_64)
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 31.35 GiB
Name: shorty
ID: NOTV:7M7T:6HMW:I6DG:4MPD:M7XM:TJKR:R6F4:EXVS:4UV5:BIJZ:KWI5
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): true
 File Descriptors: 158
 Goroutines: 273
 System Time: 2016-04-20T08:28:04.398341864+02:00
 EventsListeners: 0
Username: cupracer
Registry: https://index.docker.io/v1/
WARNING: No kernel memory limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
This is a single Docker instance on physical hardware.
Containers are usually run by using Docker Compose.
Docker server-side debugging is enabled!

```
DOCKER_OPTS="-D --dns 8.8.8.8 --storage-opt dm.datadev=/dev/data/dockerdata --storage-opt dm.metadatadev=/dev/data/dockermetadata --storage-opt dm.fs=xfs"
DOCKER_NETWORK_OPTIONS=""
```

**The issue:**
One of my containers needs to do a lot of DNS queries (~ 100 per second, if not even more).
Internal (lookup other containers by name): 50%
External (lookup various different domain names): 50%

This normally works fine and the performance is great, but after a while (~ 1-4 hrs) this container's DNS queries slow down massively, although the query times in the following output stay fast.
Since the container provides an Apache HTTP server with PHP enabled, it appeares unresponsive to the "outside world" because of those hanging DNS queries.

```
# cat /etc/resolv.conf
nameserver 127.0.0.11
options ndots:0
```

```
root@d04348b20152:/# time dig google.de

;; ANSWER SECTION:
google.de.              231     IN      A       216.58.201.227

;; Query time: 16 msec
;; SERVER: 127.0.0.11#53(127.0.0.11)
;; WHEN: Wed Apr 20 06:52:35 UTC 2016
;; MSG SIZE  rcvd: 54

real    0m0.021s
user    0m0.004s
sys     0m0.000s

root@d04348b20152:/# time dig google.de

;; ANSWER SECTION:
google.de.              224     IN      A       216.58.201.227

;; Query time: 15 msec
;; SERVER: 127.0.0.11#53(127.0.0.11)
;; WHEN: Wed Apr 20 06:52:42 UTC 2016
;; MSG SIZE  rcvd: 54


real    0m5.021s
user    0m0.004s
sys     0m0.000s

root@d04348b20152:/# time dig google.de

; <<>> DiG 9.9.5-9+deb8u6-Debian <<>> google.de
;; global options: +cmd
;; connection timed out; no servers could be reached

real    0m15.005s
user    0m0.004s
sys     0m0.000s
```

Internal DNS lookups (resolve MongoDB and Redis container IP's) work fine all the time, but queries to resolve external DNS names seem to get stuck most of the time. 
Restarting the affected container clears the situation again (for a while).

Furthermore, my system log contains a lot of the following. I found 104 of them today between 08:20:05 and 08:52:00. On yesterday, 478 of them were logged.

```
2016-04-20T08:20:05.669869+02:00 shorty docker[11372]: time="2016-04-20T08:20:05.669802699+02:00" level=error msg="More than 50 concurrent queries from 127.0.0.11:34244"
...
2016-04-20T08:52:00.377952+02:00 shorty docker[11372]: time="2016-04-20T08:52:00.377857662+02:00" level=error msg="More than 50 concurrent queries from 127.0.0.11:34244"
```

Pings from within this container to the DNS server 8.8.8.8 and other external addresses work fine. Dig's that use 8.8.8.8 directly instead of 127.0.0.11 work fine, too.
The Docker host does not seem to have any problems in DNS resolving or reachability.
The container is reachable, too. It's just the DNS issue which breaks my application.

Nothin special gets logged (even in Debug mode) regarding a failing DNS query. Just:

```
2016-04-20T09:32:08.257300+02:00 shorty docker[11372]: time="2016-04-20T09:32:08.257281875+02:00" level=debug msg="To resolve: www.google.de in "
2016-04-20T09:32:08.257334+02:00 shorty docker[11372]: time="2016-04-20T09:32:08.257295204+02:00" level=debug msg="To resolve: www.google in de"
2016-04-20T09:32:08.257347+02:00 shorty docker[11372]: time="2016-04-20T09:32:08.257301203+02:00" level=debug msg="To resolve: www in google.de"
2016-04-20T09:32:08.257368+02:00 shorty docker[11372]: time="2016-04-20T09:32:08.257309488+02:00" level=debug msg="Query www.google.de.[28] from 172.18.0.4:40514, forwarding to udp:8.8.8.8"
```

I believe that #22144 might be related to my described issue.

Before upgrading to Docker 1.11.0 I didn't limit Docker to "--dns 8.8.8.8", but I also had problems with issue #22081, so I'm using a single Google DNS server as a workaround.

### PR DESCRIPTION
**Pull request by aboch**
1) Vendoring:
- vishvananda/netlink 734d02c
- libnetwork  ed311d0

2) Update check-config.sh to check for optional modules needed for secure datapath

Fixes #22185
Fix for https://github.com/docker/libnetwork/issues/1247
Fixes crash at boot when xfrm modules are missing or are not autoloaded

**\- How to verify it**
For the bugs with an issue open, refer to the issue description.
For the crash, move the following file somewhere else and boot the daemon:
 `/lib/modules/$(uname -r)/kernel/net/xfrm/xfrm_user.ko`

**\- More info about the crash**
Originally reported offline by @fxdgear and @LK4D4 on  Antegros and Gentoo Linux

```
ERRO[0001] could not create netlink handle on initial namespace: protocol not supported
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xd07df7]

goroutine 1 [running]:
panic(0x1467cc0, 0xc420012100)
       /home/moroz/go/src/runtime/panic.go:500 +0x1a1
github.com/vishvananda/netlink.(*Handle).LinkByName(0x0, 0xc42085a6c0, 0xf, 0x20220c0, 0x68b249, 0x20220b8, 0x68b0d9)
       /home/moroz/project/workspace/src/github.com/docker/docker/vendor/src/github.com/vishvananda/netlink/link_linux.go:759 +0x37
github.com/docker/libnetwork/drivers/bridge.newInterface(0x0, 0xc4205f7f40, 0xc4202a0f28)
       /home/moroz/project/workspace/src/github.com/docker/docker/vendor/src/github.com/docker/libnetwork/drivers/bridge/interface.go:41 +0xb9
github.com/docker/libnetwork/drivers/bridge.(*driver).createNetwork(0xc42061c000, 0xc4205f7f40, 0x0, 0x0)
       /home/moroz/project/workspace/src/github.com/docker/docker/vendor/src/github.com/docker/libnetwork/drivers/bridge/bridge.go:628 +0x39d
github.com/docker/libnetwork/drivers/bridge.(*driver).populateNetworks(0xc42061c000, 0x5, 0x1645258)
       /home/moroz/project/workspace/src/github.com/docker/docker/vendor/src/github.com/docker/libnetwork/drivers/bridge/bridge_store.go:48 +0x1e8
```

**\- A picture of a cute animal (not mandatory but encouraged)**
gattorosso

### CODE REVIEW

**Comment by LK4D4** [comment]
Trying right now

**Comment by LK4D4** [comment]
Fixes problem for me.

File: contrib/check-config.sh [hunk 1]
```diff
@@ -249,6 +249,8 @@ echo '- Network Drivers:'
 {
 	echo '- "'$(wrap_color 'overlay' blue)'":'
 	check_flags VXLAN | sed 's/^/  /'
+	echo '  Optional:'
```
Comment (LK4D4) [inline contrib/check-config.sh]: Pls add comment why it might be needed.

Comment (aboch) [inline contrib/check-config.sh]: Thanks, will add the comment.

**Comment by LK4D4** [comment]
LGTM

**Comment by aboch** [comment]
Janky's failure looks unrelated to the code changes.

**Comment by jfrazelle** [comment]
looks good to me

**Comment by vdemeester** [comment]
LGTM 🐯

**Pull request merged**